### PR TITLE
fix: set CRT log level without fatalError

### PR DESCRIPTION
**Pull request by jbelkins**
## Issue \#
https://github.com/awslabs/aws-sdk-swift/issues/1761

## Description of changes
Fixes a crash that happens immediately upon setting the CRT log level.
The cause of the crash is that since `setLogLevel()` is currently an instance function, the logger has to be crated (as part of the shared instance) before it can be 

`SDKDefaultIO.setLogLevel()` is changed from an instance method to a static method; this is technically a breaking change but it has no effect since the current API crashes immediately when used.

## Scope
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/ClientRuntime/Networking/Http/CRT/SDKDefaultIO.swift [hunk 1]
```diff
@@ -25,13 +19,24 @@ public final class SDKDefaultIO: @unchecked Sendable {
...
     }
 
     private init() {
         CommonRuntimeKit.initialize()
-        Self.setupLogger(level: .none)
+        do {
+            try Logger.initialize(target: .standardOutput, level: .none)
+        } catch CommonRunTimeError.crtError(let error)
+                    where error.code == 6 && error.name == "AWS_ERROR_UNSUPPORTED_OPERATION" {
+            // logger was already initialized, no need to initialize it
```
Comment (jbelkins) [inline Sources/ClientRuntime/Networking/Http/CRT/SDKDefaultIO.swift:36]: Catch the "logger already initialized" error and ignore it if the logger was already initialized when the log level was set.

File: Sources/ClientRuntime/Networking/Http/CRT/SDKDefaultIO.swift [hunk 2]
```diff
@@ -6,12 +6,6 @@
 //
 
 import AwsCommonRuntimeKit
-import class Foundation.ProcessInfo
-#if os(Linux)
-import Glibc
-#else
-import Darwin
-#endif
```
Comment (jbelkins) [inline Sources/ClientRuntime/Networking/Http/CRT/SDKDefaultIO.swift:14]: The deleted imports above were unused.

File: Sources/ClientRuntime/Networking/Http/CRT/SDKDefaultIO.swift (hunk 1 above)
Comment (sichanyoo) [reply Sources/ClientRuntime/Networking/Http/CRT/SDKDefaultIO.swift:36]: Does this mean the log level can't be changed after it's been initialized?

Comment (sichanyoo) [reply Sources/ClientRuntime/Networking/Http/CRT/SDKDefaultIO.swift:36]: Discussed offline, this is due to design of CRT logger

**Review by dayaffe — APPROVED** [review]

**Pull request merged**